[changjy81] iP

changjy81 opened the pull request:
# Astraea
> The whimsy of the gods are not to be underestimated.  

A bored deity offers to help track your mortal tasks. :dizzy:
- Create or delete Todo, Deadline or Event tasks 
- Mark tasks as completed or incomplete
- It is watching and judging you. Always.

## Features
- [x] Todo tasks with name and completion status
- [x] Deadline tasks with name, deadline and completion status
- [x] Event tasks with name, start/end time and completion status
- [x] Search through tasks by name
- [ ] GUI (work in progress)

## Installation
1. Check the [Releases](https://github.com/changjy81/ip/releases) page and download the `astraea.jar` file from the **latest** release.
2. Place the `astraea.jar` file in a folder of your choosing. Note that other folders will be created in where you run the program.
3. Run `astraea.jar`.

### Obligatory Code Block
```java
public static void main(String[] args) {
    new Astraea().run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

changjy81 committed:
rename project to Astraea, refactor, update intro and exit dialogue

changjy81 committed:
add indent to line, split code into functions, add echo functionality

changjy81 committed:
add indent to line, split code into functions, add echo functionality

changjy81 committed:
remove unnecessary import

changjy81 committed:
add store and list functionality, disable echo

changjy81 committed:
add Task class with isDone boolean, add mark/unmark task functionality

changjy81 committed:
add task types, narrow down input accepted

changjy81 committed:
I/O redirection testing

changjy81 committed:
add input checking and error handling

changjy81 committed:
add delete functionality

changjy81 committed:
Add saving tasks to file (no reading yet)

changjy81 committed:
Fix not printing error message when given blank name in Todo task

changjy81 committed:
Add file reading

changjy81 committed:
Merge branch 'branch-Level-7'

changjy81 committed:
Add date and datetime support when keyed in specific format

changjy81 committed:
Merge branch 'branch-Level-8'

changjy81 committed:
Add skeleton Command structure

changjy81 committed:
Add Parser, TaskList, AstraeaUI skeleton

changjy81 committed:
Add DateParser.java, Storage.java
Modify AstraeaUI.java, TaskList.java

changjy81 committed:
Command classes: Adopt OOP structure

All logic for executing user commands are currently in Astraea.java.

Let's create classes for all command types and move all the logic
related to command execution to their respective classes.

This is a major step to increase OOP-ness of Astraea.

changjy81 committed:
Parser.java: Adopt new Command structure

Current assumption of Command using CommandType to differentiate
commands and execution logic.

Let's update Parser to properly use the OOP structure done by the
previous commit.

With logic for each command only present in specialized subclasses,
the current code cannot function.

changjy81 committed:
Update AstraeaInputException.java

All error messages are being printed manually to console.

Let's update it to use the automatic formatting provided by
AstraeaUI.java.

changjy81 committed:
Update Deadline.java, Event.java

Creation of tasks during load from file is being done in
Astraea.java.

Let's create factory methods in Deadline and Event to house this.

This also allows for preservation of the existing save format.

changjy81 committed:
Astraea.java: Update to adopt OOP structure

Almost everything is being handled in Astraea.java.

Let's update to allow other classes to handle everything.

Now that the OOP structure is complete, all logic can be removed
from Astraea.java and proper OOP structure can be established.

changjy81 committed:
Update AstraeaFIleException.java

While not currently used, let's update to reflect the new system for
printing error messages.

changjy81 committed:
Storage.java: Split wildcard import to single-lines

changjy81 committed:
Move all classes into packages

All classes are currently in default package.

Let's create new packages and move them into their respective
packages, adding imports as necessary.

changjy81 committed:
Merge branch 'branch-MoreOOP'

changjy81 committed:
Merge branch 'add-gradle-support'

changjy81 committed:
Add Gradle and JUnit tests for Parsers

changjy81 committed:
Add Javadoc comments to public classes and methods

changjy81 committed:
Fix codestyle

Current, import statements are out of order, switch case indentation
is incorrect, and array initialization indentation is incorrect.

Let's fix these to match with the required project codestyle.

changjy81 committed:
Add FindCommand.java

changjy81 committed:
Add FIND to CommandType enum

changjy81 committed:
Add parse support for FindCommand and corresponding exception

changjy81 committed:
Merge branch 'branch-A-JavaDoc'

changjy81 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/astraea/command/Command.java
#	src/main/java/astraea/command/DeleteCommand.java
#	src/main/java/astraea/command/EventCommand.java
#	src/main/java/astraea/command/ListCommand.java
#	src/main/java/astraea/command/TodoCommand.java
#	src/main/java/astraea/command/ToggleCommand.java
#	src/main/java/astraea/storage/Storage.java
#	src/main/java/astraea/task/Deadline.java
#	src/main/java/astraea/task/Event.java

changjy81 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/astraea/exception/AstraeaInputException.java
#	src/main/java/astraea/parser/Parser.java

changjy81 committed:
Fix Javadoc style

changjy81 committed:
Fix abbreviation case in AstraeaUi

changjy81 committed:
Add Javadoc header to Astraea class

changjy81 committed:
Add header Javadoc to Command and Task constructors

changjy81 committed:
Merge branch 'branch-A-CheckStyle'

changjy81 committed:
Add JavaFX support

Program currently runs by reading input from console and printing
to console.

Let's attempt to create a friendlier user interface with JavaFX.

changjy81 committed:
Merge branch 'branch-Level-10'

hhlh27 commented on src/main/java/astraea/parser/DateParser.java:
```diff
@@ -0,0 +1,44 @@
+ * Class containing static methods associated with parsing Strings as LocalDate or LocalDateTime.
+ */
+public class DateParser {
+    /**
```
Good usage of Javadoc comments

hhlh27 commented on src/main/java/astraea/parser/Parser.java:
```diff
@@ -0,0 +1,172 @@
+        }
+    }
+
+    private static String[] processFindTokens(String[] tokens) throws AstraeaInputException {
```
Method and variable names are descriptive and meaningful

hhlh27 commented on src/main/java/astraea/Astraea.java:
```diff
@@ -0,0 +1,111 @@
+                Thread.currentThread().interrupt(); // Restore interrupted status
+            }
+            Platform.runLater(Platform::exit); // Ensure it runs on JavaFX thread
+        }).start();
```
Code is well-structured, separating concerns between UI, logic, and task management

hhlh27 commented on src/main/java/astraea/command/Command.java:
```diff
@@ -0,0 +1,71 @@
+     */
+    public Command(CommandType commandType, String[] args) {
+        this.commandType = commandType;
+        this.args = args;
```
Perhaps a validation to ensure args is not null or empty?

hhlh27 reviewed (commented):
Looks good overall👍

changjy81 committed:
Add assert checks in Storage.java

Storage implicitly requires the information to be stored or read
are present and not null.

This may cause issues if these public methods are called
inappropriately.

Let's add some checks to make debugging such potential issues
easier.

changjy81 committed:
Extract saving new Task with error handling

Task-related Command subclasses share the same duplicated code
fragment to save the new Task to file and return the same
error message if an exception occurs.

Extracting this duplicated code improves code quality.

Let's extract and rewrite this code to fit within the Storage class,
allowing the relevant code in each Command subclass to be just
two/three lines.

This implementation is also another step towards disabling the
console-based UI to be obsoleted later.

changjy81 committed:
Remove AstraeaUI.java

The AstraeaUI contains methods used for printing formatted messages
to console.

As the JavaFX GUI was introduced in previous commits, the console
UI is no longer required.

Let's remove AstraeaUI and all its usages to reduce code duplication
and complexity.

The usage of ExitCommand from "bye" will also need to be
reconsidered, since the GUI window can be closed to
terminate the program.

changjy81 committed:
Extract line parse method from read method

changjy81 committed:
Rewrite read method in Storage.java to use Streams

changjy81 committed:
Merge pull request #1 from changjy81/branch-A-Assertions

Add assert checks in Storage.java

changjy81 committed:
Merge branch 'master' into branch-A-CodeQuality

changjy81 committed:
Merge branch 'master' into branch-A-Streams

changjy81 committed:
Merge pull request #2 from changjy81/branch-A-CodeQuality

Improve code quality

changjy81 committed:
Merge branch 'master' into branch-A-Streams

changjy81 committed:
Merge pull request #3 from changjy81/branch-A-Streams

Rewrite read method in Storage.java to use Streams

changjy81 committed:
Add alias support

Commands restrict the first word of user input to a predefined
range to be considered valid inputs.

Users may want to use alternative names for these commands.

Let's add support for creating/removing aliases of commands,
as well as saving the created aliases to a file.

changjy81 committed:
Merge branch 'branch-C-FriendlierSyntax'

changjy81 committed:
Reformat long methods and remove dead code

changjy81 committed:
Move String[] commands constant from Alias.java to Command.java

changjy81 committed:
Merge branch 'branch-A-AiAssisted'

changjy81 committed:
Update unit testing of Parser for new commands

changjy81 committed:
Add black borders to JavaFX GUI

changjy81 committed:
Add check for pipe characters in Parser.java

changjy81 committed:
Fix incorrect DateTime display format

changjy81 committed:
Set JavaFX application window title

changjy81 committed:
Add product screenshot

changjy81 committed:
Update user guide

changjy81 committed:
Change the image used for responses from Astraea

changjy81 committed:
Fix ExitCommand and update user guide

changjy81 committed:
Update user guide to include warning against using pipe character
